core/src/main/scala/org/broadinstitute/dsde/rawls/util/WorkspaceSupport.scala

```diff
@@ -0,0 +1,68 @@
+package org.broadinstitute.dsde.rawls.util
+
+import akka.http.scaladsl.model.StatusCodes
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{DataAccess, ReadWriteAction}
+import org.broadinstitute.dsde.rawls.dataaccess.{SamDAO, SlickDataSource, SlickWorkspaceContext}
+import org.broadinstitute.dsde.rawls.model.{ErrorReport, SamResourceAction, SamResourceTypeNames, SamWorkspaceActions, UserInfo, Workspace, WorkspaceAttributeSpecs, WorkspaceName}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait WorkspaceSupport {
```

Comment:
at some point i'd love to have `WorkspaceService` use this but in the interest of not cluttering something that will be a feature branch, i'll TODO it for after this stuff is merged (or i might factor it out into it's own PR, merge that, and then rebase this stuff)

Reply:
I like the idea of making it its own PR then rebasing this one ... I worry about the duplicated code existing for too long